public/js/redux/chartsRedux.js

```diff
@@ -1,28 +1,55 @@
 import { State, Actions, Effect } from 'jumpstate';
-import getChartData from 'services/getChartData';
+import api from 'services/Api';
 import { createYTotalsList, createPartialsList, formattedSeries } from 'helpers/chartsHelpers';
 
 /* ------------- State Management ------------- */
 
 Effect('filterDesk', (filterObj) => {
+    Actions.resetErrors();
     Actions.updateFilter(filterObj);
     Actions.toggleIsUpdatingCharts(true);
-    getChartData(filterObj).then(chartsData => {
-        Actions.toggleIsUpdatingCharts(false);
-        Actions.updateComposerVsIncopy(chartsData);
-    });
+    api.getComposerVsIncopy(filterObj.startDate, filterObj.endDate, filterObj.desk)
+        .then(composerVsInCopyData => {
+            Actions.toggleIsUpdatingCharts(false);
```

Comment:
We chatted a bit about this yesterday - Had a bit of a thought about a good candidate for a hook:

```javascript
const updatingChartHook = Hook(action => {
// Not exactly the most solid of checks, and may need 'includes' polyfill
  if (action.type.includes('update') || action.type.includes('Failed') ) { 
    Actions.toggleIsUpdatingCharts(false)
  }
})
```

Then the other bit get cleaner:

```javascript
api[`get${chart}`](filterObj.startDate, filterObj.endDate, filterObj.desk)
        .then(Actions[`update${chart}`])
        .catch(Actions[`get${chart}Failed`]);
```